### ISSUE TITLE
Problem: Runtime-added jobs run multiple times from app_start instead of job registration

### ISSUE THREAD

**Issue by collinmurray** [quantum-elixir/quantum-core#413]
**EDIT: Resolved by switching to master branch. Problem still persists on 2.3.4 release.**

### Description

When loading jobs at runtime the scheduler retroactively runs the job multiple times based on how long it's been since the start of the application.
ie: 

1) Start app up at 00:00:00 without any jobs
2) add_job at 00:05:00 (5 minutes later) with an interval of 1 minute `(~e[* * * * *])`.
3) The job is run immediately but 4 other instances of the job are queued and run at the same time.

This is a follow-up bug related to the discussion here: https://github.com/quantum-elixir/quantum-core/issues/269#issuecomment-342842180

### Setup
- run `mix phx.new hello`
- added deps to mix.exs
- Created scheduler.ex with the contents from README.md
- Added scheduler to application.ex
- DID NOT add any jobs to `config/config.exs`

```
Erlang/OTP 21 [erts-10.3.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
IEx 1.8.1 (compiled with Erlang/OTP 21)

"quantum": {:hex, :quantum, "2.3.4",...
```

### Reproducing
So now we have an app correctly configured with the Quantum scheduler ready for use.

I start up the application with `iex -S mix` and get:
```
[debug] [:nonode@nohost][Elixir.Quantum.JobBroadcaster] Loading Initial Jobs from Config
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Unknown last execution time, using now
```

- run `import Crontab.CronExpression`
- run 
```
Hello.Scheduler.new_job() \
|> Quantum.Job.set_name(:new_job) \
|> Quantum.Job.set_schedule(~e[* * * * *]e) \
|> Quantum.Job.set_task(fn -> IO.puts("WORKING JOB!") end) \
|> Hello.Scheduler.add_job()
```

Note that we're using a 1 second interval here so I can produce quick output. The same happens at a 1m interval.
What we see in the **Output** section below is the job running once for every second since the start of the application (18+ times) followed by 1 every second. 

This happens regardless of whether we have jobs in config.ex or not and will happen on registration of a job with any interval. ie: After you app has been running for a week, if you add a job to run every day you can expect it to run 7 times as you register it which isn't the intended effect.

### Output
```
collinm$ iex -S mix
Erlang/OTP 21 [erts-10.3.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

[info] [swarm on nonode@nohost] [tracker:init] started
[debug] [:nonode@nohost][Elixir.Quantum.JobBroadcaster] Loading Initial Jobs from Config
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Unknown last execution time, using now
Interactive Elixir (1.8.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> import Crontab.CronExpression
Crontab.CronExpression
iex(2)> [info] [swarm on nonode@nohost] [tracker:cluster_wait] joining cluster..
[info] [swarm on nonode@nohost] [tracker:cluster_wait] no connected nodes, proceeding without sync
Timex.now()
#DateTime<2019-07-05 12:20:40.903175Z>
iex(3)> Timex.now()
#DateTime<2019-07-05 12:20:47.910877Z>
iex(4)> Hello.Scheduler.new_job() \
...(4)> |> Quantum.Job.set_name(:new_job) \
...(4)> |> Quantum.Job.set_schedule(~e[* * * * *]e) \
...(4)> |> Quantum.Job.set_task(fn -> IO.puts("WORKING JOB!") end) \
...(4)> |> Hello.Scheduler.add_job()
:ok
[debug] [:nonode@nohost][Elixir.Quantum.JobBroadcaster] Adding job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Adding job :new_job
iex(5)> [debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:33)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:34)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:35)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:36)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:37)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:38)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:39)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:40)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:41)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:42)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:43)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:44)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:45)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:46)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:47)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:48)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:49)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting ASAP (2019-07-05T12:20:50)
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting at -576460732385 (2019-07-05T12:20:51)
WORKING JOB!
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting at -576460731385 (2019-07-05T12:20:52)
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting at -576460730385 (2019-07-05T12:20:53)
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting at -576460729385 (2019-07-05T12:20:54)
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Continuing Execution Broadcasting at -576460728385 (2019-07-05T12:20:55)
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok

BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
       (v)ersion (k)ill (D)b-tables (d)istribution
^C
```

**Comment by collinmurray** [comment]
UPDATE: Looks like this has already been resolved in the master branch! https://github.com/quantum-elixir/quantum-core/pull/358

Is it possible to find out when this fix is going to be available as a release?

```
collinm$ iex -S mix
Erlang/OTP 21 [erts-10.3.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

[debug] [:nonode@nohost][Elixir.Quantum.JobBroadcaster] Loading Initial Jobs from Config
Interactive Elixir (1.8.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> import Crontab.CronExpression
Crontab.CronExpression
iex(2)> Hello.Scheduler.new_job() \
...(2)> |> Quantum.Job.set_name(:new_job) \
...(2)> |> Quantum.Job.set_schedule(~e[* * * * *]e) \
...(2)> |> Quantum.Job.set_task(fn -> IO.puts("WORKING JOB!") end) \
...(2)> |> Hello.Scheduler.add_job()
[debug] [:nonode@nohost][Elixir.Quantum.JobBroadcaster] Adding job :new_job
:ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Adding job :new_job
iex(3)> [debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
WORKING JOB!
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execution ended for job :new_job, which yielded result: :ok
[debug] [:nonode@nohost][Elixir.Quantum.ExecutionBroadcaster] Scheduling job for execution :new_job
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Task for job :new_job started on node :nonode@nohost
[debug] [:nonode@nohost][Elixir.Quantum.Executor] Execute started for job :new_job
WORKING JOB!
```

**Comment by maennchen** [comment]
@collinmurray Thanks for the report.

I‘ll not be able to release anything from master soon. The master is in a very early stage for the next major. It‘ll get a lot more refactorings and a new distribution method before this will be available.

Do you have time to look for a PR based on v2? I‘ll do a fix eventually, but this may take some time since I‘m quite absorbed in my work currently.

**Comment by collinmurray** [comment]
Hey @maennchen 

Thanks for the quick reply, I'll continue to point to master for now as it resolves me issue.

I definitely have to review a PR. Please link it here and I'll go through it. :+1:

**Comment by maennchen** [comment]
Fixed on master, will be released with #427
